localstripe/resources.py

```diff
@@ -2251,6 +2306,9 @@ def _update(self, metadata=None, items=None, tax_percent=None,
         # To return 404 if not existant:
         if items[0]['tax_rates'] is not None:
             [TaxRate._api_retrieve(tr) for tr in items[0]['tax_rates']]
+        elif default_tax_rates is not None:
+            # To return 404 if not existant:
+            [TaxRate._api_retrieve(tr) for tr in default_tax_rates]
```

Comment:
Why this code? If it's for the "404" check, it's already done 13 lines above.

But I agree the code could be more readable here, to do so, we could move the block `if tax_percent is not None: ... if default_tax_rates is not None: ...` below?
```diff
         except AssertionError:
             raise UserError(400, 'Bad request')
 
         if items is None or len(items) != 1 or not items[0]['plan']:
             raise UserError(500, 'Not implemented')
 
         # To return 404 if not existant:
         new_plan = Plan._api_retrieve(items[0]['plan'])
 
+        if tax_percent is not None:
+            self.tax_percent = tax_percent
         # To return 404 if not existant:
+        if default_tax_rates is not None:
+            self.default_tax_rates = [TaxRate._api_retrieve(tr)
+                                      for tr in default_tax_rates]
         if items[0]['tax_rates'] is not None:
             [TaxRate._api_retrieve(tr) for tr in items[0]['tax_rates']]
 
         self.quantity = items[0]['quantity']
```

Reply:
Hum you are right.
I think I add this code each time I found the equivalent on (depreciated) `tax_rates` .
Here is the diff applied on the last commit:
```patch
--- a/localstripe/resources.py
+++ b/localstripe/resources.py
@@ -2296,23 +2296,20 @@ class Subscription(StripeObject):
         except AssertionError:
             raise UserError(400, 'Bad request')
 
-        if tax_percent is not None:
-            self.tax_percent = tax_percent
-        if default_tax_rates is not None:
-            self.default_tax_rates = [TaxRate._api_retrieve(tr)
-                                      for tr in default_tax_rates]
-
         if items is None or len(items) != 1 or not items[0]['plan']:
             raise UserError(500, 'Not implemented')
 
         # To return 404 if not existant:
         new_plan = Plan._api_retrieve(items[0]['plan'])
+
+        if tax_percent is not None:
+            self.tax_percent = tax_percent
+        if default_tax_rates is not None:
+            self.default_tax_rates = [TaxRate._api_retrieve(tr)
+                                      for tr in default_tax_rates]
         # To return 404 if not existant:
         if items[0]['tax_rates'] is not None:
             [TaxRate._api_retrieve(tr) for tr in items[0]['tax_rates']]
-        elif default_tax_rates is not None:
-            # To return 404 if not existant:
-            [TaxRate._api_retrieve(tr) for tr in default_tax_rates]
 
         self.quantity = items[0]['quantity']
```
Thanks!